Sources/Rendering/Core/CubeAxesActor/index.js

```diff
@@ -0,0 +1,831 @@
+      );
+    }
+
+    const changeDetected = model.axisLabels.some(
+      (item, index) => item !== array[index]
+    );
+
+    if (changeDetected || model.axisLabels.length !== array.length) {
+      model.axisLabels = [].concat(array);
+      publicAPI.update(true);
+      console.log('forcing update');
+      publicAPI.modified();
+      return true;
+    }
+    return false;
+  };
+
+  publicAPI.setTickTextStyle = (tickStyle) => {
+    model.tickTextStyle = { ...model.tickTextStyle, ...tickStyle };
+    publicAPI.update(true);
+    publicAPI.modified();
+  };
+
+  publicAPI.setAxisTextStyle = (axisStyle) => {
+    model.axisTextStyle = { ...model.axisTextStyle, ...axisStyle };
+    publicAPI.update(true);
+    publicAPI.modified();
+  };
+}
+
+// ----------------------------------------------------------------------------
+// Object factory
+// ----------------------------------------------------------------------------
+
+function defaultValues(initialValues) {
+  return {
+    camera: null,
+    dataBounds: [...vtkBoundingBox.INIT_BOUNDS],
+    gridLines: true,
+    axisLabels: null,
+    axisTitlePixelOffset: 35.0,
+    axisTextStyle: {
+      fontColor: 'white',
+      fontStyle: 'normal',
+      fontSize: 18,
+      fontFamily: 'serif',
+    },
+    tickLabelPixelOffset: 12.0,
+    tickTextStyle: {
+      fontColor: 'white',
+      fontStyle: 'normal',
+      fontSize: 14,
+      fontFamily: 'serif',
+    },
+    ...initialValues,
+  };
+}
+
+// ----------------------------------------------------------------------------
+
+export function extend(publicAPI, model, initialValues = {}) {
+  Object.assign(model, defaultValues(initialValues));
+
+  // Inheritance
+  vtkActor.extend(publicAPI, model, initialValues);
+
+  // internal variables
+  model.lastSize = [800, 800];
+  model.lastFacesToDraw = [false, false, false, false, false, false];
+  model.axisLabels = ['X-Axis', 'Y-Axis', 'Z-Axis'];
+  model.tickCounts = [];
+  model.textValues = [];
+  model.lastTickBounds = [];
+
+  model.mapper = vtkMapper.newInstance();
+  model.polyData = vtkPolyData.newInstance();
+  model.mapper.setInputData(model.polyData);
+  model.actor = vtkActor.newInstance();
+  model.actor.setMapper(model.mapper);
+
+  model.actor.setProperty(publicAPI.getProperty());
+  publicAPI.getProperty().setDiffuse(0.0);
+  publicAPI.getProperty().setAmbient(1.0);
+
+  model.textPolyData = vtkPolyData.newInstance();
+
+  // for texture atlas
+  model.tmPolyData = vtkPolyData.newInstance();
+  model.tmMapper = vtkMapper.newInstance();
+  model.tmMapper.setInputData(model.tmPolyData);
+  model.tmTexture = vtkTexture.newInstance();
+  model.tmTexture.setInterpolate(false);
+  model.tmActor = vtkActor.newInstance();
+  model.tmActor.setMapper(model.tmMapper);
+  model.tmActor.addTexture(model.tmTexture);
+  model.tmCanvas = document.createElement('canvas');
+  model.tmContext = model.tmCanvas.getContext('2d');
+  model.tmAtlas = new Map();
```

Comment:
Ideally we want the model to be serializable. You may want to keep your `Map()` as private (const inside closure).

Reply:
Hmm can subclasses access this internal state? e.g. how should I declare things I want to be protected in C++ terminology?

Reply:
What ever you bind to model is ~"protected"

Reply:
OK then most/all of these should stay in model right, as a subclass will typically need to access them to add or change much functionality

Reply:
Going to merge this and then we can tweak more if we want.